src/common/extensionMessaging.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import * as http from "http";
+
+/**
+ * Message server parameters.
+ */
+export let ServerDefaultParams = {
+    PORT: 8099,
+    HOST: "127.0.0.1"
+};
+
+/**
+ * Defines the messages sent to the extension.
+ * Add new messages to this enum.
+ */
+export enum ExtensionMessage {
+    START_PACKAGER,
+    STOP_PACKAGER,
+    PREWARM_BUNDLE_CACHE
+}
+
+/**
+ * Sends messages to the extension.
+ */
+export class ExtensionMessageSender {
+
+    public sendMessage(message: ExtensionMessage, args?: any[], port?: number): Q.Promise<any> {
```

Comment:
I updated the handler type to take parameters instead of an array. This way, whatever you send to the server should match the arguments on the message handler.